### ISSUE TITLE
Fails to compile without AOT (enable-minimal=profiler,debug,aot)

### ISSUE THREAD

**Issue by hmargies** [mono/mono#8430]
Mono fails to compile when AOT is disabled (enable-minimal=profiler,debug,aot).

## Steps to Reproduce

```
sh autogen.sh
./configure --enable-minimal=profiler,debug,aot
make
```

## Current Behavior

The function ```mono_aot_get_weak_field_indexes``` is used in mini-runtime.c and implemented in ```aot-runtime.c``` but not compiled in due to ```DISABLE_AOT``` being defined.

```
  CC       mono_boehm-main.o
  CCLD     mono-boehm
./.libs/libmini.a(libmini_la-mini-runtime.o): In function `mini_init':
/home/henry/Development/test/mono/mono/mini/mini-runtime.c:4041: undefined reference to `mono_aot_get_weak_field_indexes'
collect2: error: ld returned 1 exit status
make[4]: *** [mono-boehm] Error 1
make[4]: Leaving directory `/home/henry/Development/test/mono/mono/mini'
make[3]: *** [all] Error 2
make[3]: Leaving directory `/home/henry/Development/test/mono/mono/mini'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/home/henry/Development/test/mono/mono'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/henry/Development/test/mono'
make: *** [all] Error 2
```

## On which platforms did you notice this

[ ] macOS
[X ] Linux
[ ] Windows

**Version Used**: 5.10.1.47 and also master from GitHub

**Comment by mikaelj** [comment]
Any progress?

**Comment by drgroot** [comment]
Seem to have the same problem with version 5.12.0.226 on Linux.
Reopen?

**Comment by lewurm** [comment]
it wasn't backported to the release branches.

@marek-safar @luhenry is it okay to backport it to 2018-02 still? https://github.com/mono/mono/pull/9394

**Comment by jbatt33** [comment]
Not sure if this is fixed in 5.16 release against ARM builds.  x86 appears to build fine with AOT disabled, but receiving the following error from the ARM build:

"
mono_aot_get_ftnptr_arg_trampoline
"

```

/libmonogc.a/stubborn.o .libs/libmonoboehm-2.0.lax/libmonogc.a/typd_mlc.o .libs/libmonoboehm-2.0.lax/libmonogc.a/win32_threads.o
| /usr/src/debug/mono/5.16.0.179-r0/mono-5.16.0.179/mono/mini/mini-trampolines.c:163: error: undefined reference to 'mono_aot_get_ftnptr_arg_trampoline'
| collect2: error: ld returned 1 exit status
| arm-angstrom-linux-gnueabi-ar: `u' modifier ignored since `D' is the default (see `U')
| Makefile:1398: recipe for target 'mono-boehm' failed
| make[4]: *** [mono-boehm] Error 1
| make[4]: *** Waiting for unfinished jobs....
| arm-angstrom-linux-gnueabi-libtool: link: (cd .libs/libmonosgen-2.0.lax/libeglib.a && arm-angstrom-linux-gnueabi-ar x "/opt/yocto/angstrom/build/tmp-angstrom-glibc/work/armv7at2hf-neon-angstrom-linux-gnueabi/mono/5.16.0.179-r0/mono-5.16.0.179/mono/mini/../../mono/eglib/.libs/libeglib.a")
| /usr/src/debug/mono/5.16.0.179-r0/mono-5.16.0.179/mono/mini/mini-trampolines.c:163: error: undefined reference to 'mono_aot_get_ftnptr_arg_trampoline'
| collect2: error: ld returned 1 exit status
| Makefile:1402: recipe for target 'mono-sgen' failed
| make[4]: *** [mono-sgen] Error 1
| arm-angstrom-linux-gnueabi-libtool: link: arm-angstrom-linux-gnueabi-ar cru .libs/libmonosgen-2.0.a   .libs/libmonosgen-2.0.lax/libmini.a/libmini_la-abcremoval.o .libs/libmonosgen-2.0.lax/libmini.a/libmini_la-alias-analysis.o .libs/libmonosgen-2.0.lax/libmini.a/libmini_la-aot-compiler.o .libs/libmonosgen-2.0.lax/libmini.a/libmini_la-aot-
```

What other information can I provide?
